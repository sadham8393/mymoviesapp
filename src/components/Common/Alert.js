import React from 'react';
import PropTypes from 'prop-types';

const Alert = (props) =>{
    return (
        <nav className = {`${props.alertClass} alert-container`}>
            <div className="nav-wrapper">
            <ul className="left">
                <h5>{props.alertMessage}</h5>
            </ul>
            <ul className="right">
                <i className="small material-icons alert-close-btn " onClick = {props.alertClose}> close </i>
            </ul>
            </div>
        </nav>
    )
}

Alert.propTypes = { 
    alertClass: PropTypes.string, 
    alertMessage: PropTypes.string,
    alertClose: PropTypes.func
} 

export default Alert;