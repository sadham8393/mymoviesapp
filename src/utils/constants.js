// URl constants

export const MOVIEDB_PATH = "https://api.themoviedb.org/3/";
export const REACT_MOVIEDB_API_KEY  = "?api_key=0e89680c8b3fa0f259b53bbcc8f6091d";
export const MOVIEDB_IMAGE_APPEND_URL = "https://image.tmdb.org/t/p/w500/";
export const TRENDING_DAY_URL = "trending/all/day";


export const NO_IMAGE_URl = "https://s3-ap-southeast-1.amazonaws.com/upcode/static/default-image.jpg";

//Common Constants
export const INPUT_TYPE_TEXT = "text";
export const INPUT_TYPE_SELECT = "select";
export const NOT_AVAILABLE = "Not Available";


//Action Constants
export const FETCH_TRENDING_MOVIES = "FETCH_TRENDING_MOVIES";
export const FETCH_TV_MOVIES = "FETCH_TV_MOVIES";
export const DELETE_TV_MOVIES = "DELETE_TV_MOVIES";
export const MOVIES_FETCHED_SUCCESS = "MOVIES_FETCHED_SUCCESS";
export const MOVIES_FETCHED_ERROR = "MOVIES_FETCHED_ERROR";
export const DELETE_MOVIE_SUCCESS = "DELETE_MOVIE_SUCCESS";
export const DELETE_MOVIE_FAILURE = "DELETE_MOVIE_FAILURE";

export const MOVIES_FETCHED_NO_RESULTS = "MOVIES_FETCHED_NO_RESULTS";

export const RESET_RESPONSE = "RESET_RESPONSE";

export const NO_RESULTS_FOUND = "No results found for the given search input";

export const SERVER_UNAVAILABE = "Server unavailable. Please try after sometime"



