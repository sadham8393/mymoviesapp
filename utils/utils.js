import configureStore from 'redux-mock-store'

export const findByTestClass = (component, className) => {
    const wrapper = component.find(`${className}`);
    return wrapper;
};

export const findByComponent = (component, innerComponent) => {
    const wrapper = component.find(innerComponent);
    return wrapper;
};

export const testStore = (initialState) => {
    let store;
    const mockStore = configureStore();
    store = mockStore(initialState)
    return store;
};