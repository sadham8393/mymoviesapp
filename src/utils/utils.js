import { NOT_AVAILABLE } from './constants';

/**
 * Convert date format from "yyyy-mm-dd" to "dd/mm/yy"
 */

export function convertDate(date){
    if(date){
        return date.toString().split("-").reverse().join("/");
    }
    return NOT_AVAILABLE;
}