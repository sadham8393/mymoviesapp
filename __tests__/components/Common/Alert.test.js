import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { findByTestClass } from '../../../utils/utils';
import Alert from '../../../src/components/Common/Alert';

describe('<Alert />', () => { 
    let wrapper;
    let testProps;
    let mockFunc;
    beforeEach(() => {
        mockFunc = jest.fn();
        testProps = {
            alertClass : "show info",
            alertClose : mockFunc
        };
        wrapper = shallow(<Alert {...testProps} />);
    });
    it('should render correctly', () => {
        const tree = renderer.create(<Alert {...testProps} />);
        expect(tree.toJSON()).toMatchSnapshot();
    });
    it('should expect nav-wrapper to be 1', () => {
        const component = findByTestClass(wrapper, '.nav-wrapper');
        expect(component.length).toBe(1);
    });
   
    it('should render ul left align correctly and to be 1', () => {
        const component = findByTestClass(wrapper, 'ul.left');
        expect(component.length).toBe(1);
    });

    it('should render ul right align correctly and to be 1', () => {
        const component = findByTestClass(wrapper, 'ul.right');
        expect(component.length).toBe(1);
    });

    it('should render close button correctly', () => {
        const component = findByTestClass(wrapper, '.alert-close-btn');
        expect(component.length).toBe(1);
    });

    it('Close should emit callback on click event', () => {
        const button = findByTestClass(wrapper, '.alert-close-btn');
        button.simulate('click');
        const callback = mockFunc.mock.calls.length;
        expect(callback).toBe(1);
    });
    
});