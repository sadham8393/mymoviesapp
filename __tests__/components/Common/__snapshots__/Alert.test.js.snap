// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should render correctly 1`] = `
<nav
  className="show info alert-container"
>
  <div
    className="nav-wrapper"
  >
    <ul
      className="left"
    >
      <h5 />
    </ul>
    <ul
      className="right"
    >
      <i
        className="small material-icons alert-close-btn "
        onClick={[MockFunction]}
      >
         close 
      </i>
    </ul>
  </div>
</nav>
`;
